src/transaction/log_manager.c

```diff
@@ -10214,6 +10426,3742 @@ logtb_tran_update_stats_online_index_rb (THREAD_ENTRY * thread_p, void *data, vo
+
+  LSA_SET_NULL (&cdc_Gl.next_lsa);
+
+  /* *INDENT-OFF* */
+  //cubthread::looper looper = cubthread::looper (cdc_log_producer_interval);
+  cubthread::looper looper = cubthread::looper (std::chrono::milliseconds (10));
+  cubthread::entry_callable_task *daemon_task = new cubthread::entry_callable_task (cdc_log_producer_execute);
+
+  cdc_log_Producer_daemon = cubthread::get_manager ()->create_daemon (looper, daemon_task, "cdc_log_producer"); 
+  /* *INDENT-ON* */
+}
+
+void
+cdc_thread_checker_daemon_init ()
+{
+  assert (cdc_Thread_checker_daemon == NULL);
+
+  /* *INDENT-OFF* */
+  cubthread::looper looper = cubthread::looper (std::chrono::seconds (1));
+  cubthread::entry_callable_task *daemon_task = new cubthread::entry_callable_task (cdc_thread_checker_execute);
+
+  cdc_Thread_checker_daemon = cubthread::get_manager ()->create_daemon (looper, daemon_task, "cdc_thread_checker"); 
+  /* *INDENT-ON* */
+}
+
+void
+cdc_daemons_init ()
+{
+#if 1				//JOOHOK
+  _er_log_debug (ARG_FILE_LINE, "cdc_daemons_init ");
+#endif
+  cdc_log_producer_daemon_init ();
+  cdc_thread_checker_daemon_init ();
+}
+
+void
+cdc_daemons_destroy ()
+{
+#if 1				//JOOHOK
+  _er_log_debug (ARG_FILE_LINE, "cdc_daemons_destroy ");
+#endif
+
+  cdc_finalize ();
+
+  /* *INDENT-OFF* */
+  cubthread::get_manager ()->destroy_daemon (cdc_log_Producer_daemon);
+  cubthread::get_manager ()->destroy_daemon (cdc_Thread_checker_daemon);
+  /* *INDENT-ON* */
+#if 1				//JOOHOK
+  _er_log_debug (ARG_FILE_LINE, "cdc_daemons_destroy done");
+#endif
+
+}
+
+void
+cdc_wakeup_log_producer ()
+{
+  if (cdc_log_Producer_daemon != NULL)
+    {
+      cdc_log_Producer_daemon->wakeup ();
+    }
+}
+
+void
+cdc_wakeup_thread_checker ()
+{
+  if (cdc_Thread_checker_daemon != NULL)
+    {
+      cdc_Thread_checker_daemon->wakeup ();
+    }
+}
+
+#endif
+
+int
+cdc_find_lsa (THREAD_ENTRY * thread_p, time_t * time, LOG_LSA * start_lsa)
+{
+  /*
+   * 1. get volume list
+   * 2. get fpage from each volume 
+   * 3. get commit/abort/ha_dummy_server_state which contains time from fpage 
+   * */
+  int begin = log_Gl.hdr.last_deleted_arv_num;
+  int end = log_Gl.hdr.nxarv_num - 1;
+  char arv_name[PATH_MAX];
+  LOG_ARV_HEADER *arv_hdr;
+  int num_arvs = end - begin;
+
+  time_t active_start_time = 0;
+  time_t archive_start_time = 0;
+  int target_arv_num = -1;
+
+  LOG_LSA ret_lsa;
+  bool is_found = false;
+
+  int error = NO_ERROR;
+
+  /*
+   * 1. traverse from the latest log volume 
+   * 2. when num_arvs > 0, no logic to handle the active log volume 
+   * 3. check condition when i = begin while finding target_arv_num 
+   */
+
+  /*AT first, compare the time in active log volume. */
+  error = get_start_point_from_file (thread_p, -1, &ret_lsa, &active_start_time);
+  if (error == ER_FAILED || error == ER_LOG_READ)
+    {
+      return error;
+    }
+  else
+    {
+      /* NO ERROR */
+      if (active_start_time != 0 && active_start_time <= *time)
```

Comment:
if active_start_time == *time, then just use ret_lsa as start_lsa.

And I think it is nice if a user could set the time as -1 (or something) then, the server returns a LSA from default action(e.g. the first LSA of the active log volume). It is my suggestion for usability.